cluster-provision/k8s/fetch-images.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+set -euo pipefail
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+function usage {
+    cat <<EOF
+Usage: $0 <k8s-cluster-dir> [source-image-list]
+
+    Fetches all images from the cluster provision source and manifests. Returns a list that is sorted and
+    without double entries.
+
+    If source-image-list is provided this is taken as an input and added to the result.
+
+EOF
+}
+
+function check_args {
+    if [ "$#" -lt 1 ]; then
+        usage
+        exit 1
+    fi
+    if [ ! -d "$1" ]; then
+        usage
+        echo "Directory $1 does not exist"
+        exit 1
+    fi
+}
+
+function main {
+    check_args "$@"
+
+    temp_file=$(mktemp)
+    trap 'rm -f "${temp_file}"' EXIT SIGINT SIGTERM
+
+    provision_dir="$1"
+    image_regex='([a-z0-9\_\.]+[/-]?)+:[a-z0-9\_\.\-]+'
+    image_regex_w_double_quotes='"?'"${image_regex}"'"?'
+    (
+        # Avoid bailing out because of nothing found in scripts part
+        set +e
+        find "$provision_dir" -type f -name '*.sh' -print0 | \
```

Comment:
Maybe we should just store a list with all the images that we want to cache instead of trying to grep them from the files. If each provider will have a file with pull urls we can easily iterate over it and source it. I suspect that anything else that we will try to do will be error prone and eventually will fail.

Reply:
@qinqon argued against this, as the alternative process would also be error prone. Then, instead  of just putting new manifests into the folder, you also would have to update the list, which the user might forget.

Reply:
I was also favouring a simple and complete list. But for as long as we fail on the sanity check if we can't extract all images for pre-pulling I don't mind too much.

Reply:
If we go back to having all images at a file thinks like bumping CNAO is more complicated, but if we are able to automatize that we don't care too much.

Reply:
Not necessarily. You would basically locally bump your manifests, run the check, it would print the new file-list (or change the file even), you commit the diff and done.

Reply:
Yeah, this was it initially.